### ISSUE TITLE
Cross out pictograms/images in the cells

### ISSUE THREAD

**Issue by arasaac-dga** [asterics/AsTeRICS-Grid#305]
Add and option on Image tab to cross out any pictogram or image overlaying a svg like we attach (the one we use in ARASAAC)
![tachado_rojo](https://github.com/asterics/AsTeRICS-Grid/assets/82500056/931b6828-2bcf-4b54-9dbe-33bf94f19b9b)

**Comment by tchjohanna** [comment]
[tchjohanna:feature-cross-out-images-305  
](https://github.com/asterics/AsTeRICS-Grid/pull/394/commits/04a720178265328be65e0c0a6605860573966952)
Hello everyone,

I hope you're all well.

I wanted to let you know about the changes I've made in response to issue #305 about crossing out pictograms/images in cells.

What I've done:
Added a crossed CSS class: I've added a CSS class to apply a large red X to the grid elements that need to be crossed out.
Modified the Vue.js template: I modified the template to dynamically apply this crossed class to the elements that need to be crossed out. The check is carried out using the isCrossed method.
Added the logic for determining which elements should be crossed out: I added a crossedElements array to the Vue component to keep track of which elements should be crossed out. I've also implemented the isCrossed method to check whether an element is part of this array.
Why these changes:
The aim is to allow users to visually mark certain pictograms or images in the grid as irrelevant or invalid by crossing them out with a large red X. This can improve the user experience by making it clearer which elements should not be used or are awaiting correction.

My concerns:
I'm still a junior and it may not be perfect. I'd be very grateful for your feedback and suggestions for improving this feature. I'm particularly interested in :

Advice on implementing the logic to determine which items should be crossed out.
Feedback on performance and state management in Vue.js with the addition of this feature.
Thank you very much for your understanding and support. I'm ready to make any necessary changes based on your feedback.

Have a great day!

**Comment by arasaac-dga** [comment]
Good evening, Johanna and welcome to the team. Sure you have done great work. About the coding we can not commented (@klues will do) but about the feature we can commented that must be an option added in this section and area of the images tab ("Setting for image search") due that must be an option that the people that configure the communicator added when is neccesary. So there is no logic that activates it automatically at the moment, only there must be an option to modify the image as well the others like plural, color, skin color,.... Anyway, perhaps being an option implemented some logic can be added in the future controlling it, for example, from a button. But at the moment, for us it is enough if you add it to "Setting for image search".

![Screenshot_127](https://github.com/asterics/AsTeRICS-Grid/assets/82500056/4eff1877-2c7e-4e15-992a-1c756a37e930)

In order to test it, we suppose that Benjamin launch it on latest.

Thanks so much for your fisrt coding for Asterics Grid,. We are happy to join new members to the team.

**Comment by tchjohanna** [comment]
Good evening everyone,

Thank you for your warm welcome and your comments. I understand the suggestion to make this feature an option in the "Image Search Settings" section. I appreciate all the advice and have made some changes based on the initial feedback.

For now, I'll let Benjamin review the code in depth and we can make any necessary improvements based on his feedback. I look forward to learning and improving the implementation accordingly.

Thanks again for your support.

**Comment by klues** [comment]
As already stated in the comment in the PR, I've once already started working on this in branch `issue#305/cross-out-images` - so it will make most sense if you would continue from there. I think the option for crossing out should be a general option in the images tab, not only for searching at the ARASAAC API. I've already this checkbox in my branch [here](https://github.com/asterics/AsTeRICS-Grid/commit/5ace157cdc037e6d6f1da60717e60cbfc168607f#diff-2982994def9475955a2f9baecf2e60c9b7feeb7445df9632c545d544f16c26c8R33).

**Comment by arasaac-dga** [comment]
Yes, we agree with @klues Cross option must a general option for any image/picture, not only for ARASAAC pictograms. Good appoinment!!!

**Comment by tchjohanna** [comment]
//////////////////////////

Guten Tag an alle,

Ich möchte mich aufrichtig für die Fehler entschuldigen, die ich in letzter Zeit bei der Verwaltung unserer Projekte gemacht habe. Mir ist klar, dass ich mit einigen Aspekten des Programmierens noch nicht ganz vertraut bin, was unseren Fortschritt beeinträchtigt haben könnte.

Ich bin zutiefst engagiert, meine Fähigkeiten zu verbessern und alles zu lernen, was nötig ist, um effektiv zu unserem Team beizutragen. Zu diesem Zweck plane ich, Zeit mit einem ehemaligen Klassenkameraden zu verbringen, der mir helfen wird, die Logik unseres Projekts besser zu verstehen und meine Entwicklungsfähigkeiten zu stärken.

Ich danke Ihnen allen für Ihre Geduld und Ihr Verständnis. Ich bin entschlossen, noch härter zu arbeiten und dafür zu sorgen, dass meine zukünftigen Beiträge die hohen Standards unseres Teams widerspiegeln. Wenn Sie Ratschläge oder Ressourcen haben, die mir helfen könnten, wäre ich für Ihre Unterstützung sehr dankbar.

Nochmals vielen Dank für Ihre Unterstützung und Ihr Verständnis.

Mit freundlichen Grüßen,

///////////////
Hola a todos,

Me gustaría disculparme sinceramente por los errores que he cometido últimamente en la gestión de nuestros proyectos. Soy consciente de que aún no me siento del todo cómodo con ciertos aspectos de la programación, lo que puede haber afectado a nuestro progreso.

Estoy profundamente comprometido a mejorar mis habilidades y aprender lo que sea necesario para contribuir eficazmente a nuestro equipo. Con este fin, he planeado pasar algún tiempo con un antiguo compañero de clase que me ayudará a comprender mejor la lógica de nuestro proyecto y a reforzar mis habilidades de desarrollo.

Gracias a todos por vuestra paciencia y comprensión. Estoy decidido a trabajar más duro y a garantizar que mis futuras contribuciones reflejen el alto nivel de nuestro equipo. Si tienen algún consejo o recurso que pueda ayudarme, les agradecería mucho su apoyo.

Gracias de nuevo por su apoyo y comprensión.

Le deseo lo mejor,

**Comment by klues** [comment]
@tchjohanna thanks for your message, but there is no need apologize. As said, we are very happy if you want to contribute your spare time for improving AsTeRICS Grid. For me the most important points for you to be able to contribute in a productive manner are:
1. focus on one thing at a time, just one issue until it's finished, then move to the next one. There is little gain if you contribute some small chunks of code to multiple issues, but nothing is finished.
2. be sure that you start from the current and updated `master` branch and create a new branch for each issue from the `master` branch, ensuring that there are only changes regarding one issue in a pull request.
3. For me it looked like you've used some AI tools in your previous pull requests. I don't think they'll help you, please try to understand the code by yourself and ask me, if you have questions about anything. Only in that way you'll get familiar with the project.

Thanks and please keep on going :+1:

**Comment by tchjohanna** [comment]
Thank you very much for your feedback and advice. I'll take good note of each point. 

Regarding the use of AI tools, I understand your point of view and I'm going to try to dive deeper into the code myself.

I'm going to revisit my algorithm courses to strengthen my understanding and confidence in software development. I've been putting code to one side for the last few months in favour of the Adobe suite, and it's time for me to get back into programming in earnest. I'm going to try to do things right and get back up to speed.

I won't hesitate to contact you if I have any questions or need clarification, because I really want to understand the project in depth and contribute effectively.

See you soon.

**Comment by willwade** [comment]
@SteveMuriithi1 take a look at @klues  branch issue#305/cross-out-images and have a chat with @klues to see if we can get this over the line
